pdata/ptrace/jsoniter.go

```diff
@@ -0,0 +1,369 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ptrace // import "go.opentelemetry.io/collector/pdata/ptrace"
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"go.opentelemetry.io/collector/pdata/internal"
+	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
+)
+
+// NewJSONIterUnmarshaler returns a model.Unmarshaler. Unmarshals from OTLP json bytes.
+func NewJSONIterUnmarshaler() Unmarshaler {
```

Comment:
Why do we need to have 2 JSON unmarshaler? Should we remove the other JSON marshaler/unmarshaler?

Reply:
It is origin intend to make JSON unmarshaller is configurable in the otlpreceiver, jsonpb is default, jsoniter is optional. 
It is also ok to remove the jsonpb unmarshaller.

Reply:
Updated.

Reply:
@bogdandrutu Please help review when convenient, thank you :)